deploy/setup/setup.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+set -e
+
+# --- Parse CLI Arguments ---
+
+for arg in "$@"; do
+  case "$arg" in
+    --cloud-init)
+      CLOUD_INIT=true
+      ;;
+    --dev)
+      SKIP_CLONE=true
+      SKIP_DEPENDENCY_INSTALLS=true
+      SKIP_LAUNCH=true
+      # Root directory above repo
+      WBS_DIR=../../..
+      ;;
+    --local)
+      LOCALHOST=true
+      ;;
+    --skip-clone)
+      SKIP_CLONE=true
+      ;;
+    --skip-deps)
+      SKIP_DEPENDENCY_INSTALLS=true
+      ;;
+    --skip-launch)
+      SKIP_LAUNCH=true
+      ;;
+    --verbose)
+      VERBOSE=true
+      ;;
+    --web)
+      USE_WEB=true
+      ;;
+  esac
+done
+
+# --- Assign Vars with Defaults ---
+
+CLOUD_INIT="${CLOUD_INIT:-false}"
+LOCALHOST="${LOCALHOST:-false}"
+SKIP_CLONE="${SKIP_CLONE:-false}"
+SKIP_DEPENDENCY_INSTALLS="${SKIP_DEPENDENCY_INSTALLS:-false}"
+SKIP_LAUNCH="${SKIP_LAUNCH:-false}"
+REPO_URL="${REPO_URL:-https://github.com/wmde/wikibase-release-pipeline.git}"
+USE_WEB="${USE_WEB:-false}"
+VERBOSE="${VERBOSE:-false}"
+WBS_DIR="${WBS_DIR:-/opt/wbs}"
+DEPLOY_DIR="${DEPLOY_DIR:-$WBS_DIR/wikibase-release-pipeline/deploy}"
+SETUP_DIR="${SETUP_DIR:-$DEPLOY_DIR/setup}"
+SCRIPTS_DIR="$SETUP_DIR/scripts"
+LOG_PATH="${LOG_PATH:-/tmp/wbs-deploy-setup.log}"
+
+# -- Setup logging --
+
+mkdir -p "$(dirname "$LOG_PATH")"
+touch "$LOG_PATH" 2>/dev/null || true
+
+log() {
```

Comment:
Have you considered reusing you `_logging.sh` script?

Reply:
This omission is on purpose and unfortunately necessary as this script is designed to be ran independently and before having the entire repo cloned. I will add a comment here to this effect, but basically the library scripts/_logging.sh code and this bit need to be manually kept in sync.